### PR TITLE
BGDIINF_SB-1482: Updated Transactional SPEC

### PR DESCRIPTION
**Pull request by ltshb**
* Some clean up in the error schema and examples
* Added ETag and Location headers to Collection
* Corrected Item POST/PUT/PATCH responses to give back the created/updated item
* Corrected the DEL of Asset/Item with the 200 OK instead of 204 No Content

NOTE: The ETag might still requires some changes based on up coming implementation.

### CODE REVIEW

File: spec/transaction/transaction.yml [hunk 1]
```diff
@@ -939,6 +839,12 @@ components:
     #       items:
     #         $ref: "#/components/schemas/link"
     #       type: array
+    itemIdUpdate:
+      description: >-
+        Item identifier (unique per collection. If it doesn't match the `featureId` in path
+        parameters, then the Item is renamed.
+      example: smr200-200-4-2019
+      type: string
```
Comment (boecklic) [inline spec/transaction/transaction.yml:847]: To be discussed whether we want/should/must support this case, same with renaming assets.

File: spec/transaction/transaction.yml [hunk 2]
```diff
@@ -743,129 +713,59 @@ components:
...
+    updateItem:
+      allOf:
+        - type: object
+          required:
+            - id
+          properties:
+            id:
+              $ref: "#/components/schemas/itemIdUpdate"
+        - $ref: "#/components/schemas/writeItem"
+    partialItem:
```
Comment (boecklic) [inline spec/transaction/transaction.yml:746]: nice solution!

**Review by boecklic — APPROVED** [review]
:+1: 
one thing we'll need to discuss with Business: do we want/should we support renaming of items/assets or not? In principle I would say yes, but potential consequences must be clear to everybody.

**Review by rebert — APPROVED** [review]
reasonable changes for me

File: spec/transaction/transaction.yml (hunk 1 above)
Comment (ltshb) [reply spec/transaction/transaction.yml:847]: I've asked Pasquale and Dave on Jira about this. I'll wait on their answer.

Comment (ltshb) [reply spec/transaction/transaction.yml:847]: I'll merge as is and if the rename needs to be removed we can create a new PR, we will anyway need to change the code.

**Pull request merged**